{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os.path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "BASE_PATH='/hackathon/Hackathon_Project_4/'\n",
    "jp=lambda x: os.path.join(BASE_PATH,x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File accession</th>\n",
       "      <th>File format</th>\n",
       "      <th>Output type</th>\n",
       "      <th>Experiment accession</th>\n",
       "      <th>Assay</th>\n",
       "      <th>Biosample term id</th>\n",
       "      <th>Biosample term name</th>\n",
       "      <th>Biosample type</th>\n",
       "      <th>Biosample life stage</th>\n",
       "      <th>Biosample sex</th>\n",
       "      <th>...</th>\n",
       "      <th>md5sum</th>\n",
       "      <th>File download URL</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Controlled by</th>\n",
       "      <th>File Status</th>\n",
       "      <th>Audit WARNING</th>\n",
       "      <th>Audit INTERNAL_ACTION</th>\n",
       "      <th>Audit NOT_COMPLIANT</th>\n",
       "      <th>Audit ERROR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENCFF000IZE</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000CQV</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>32e18ca6acd794dbbb9ce8cc1f986f10</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000IZ...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENCFF000IZF</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000CQV</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>e41f8dd0aaf9c019f6f1f055c8ea9952</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000IZ...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENCFF000WFL</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EUL</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>b91dcf11ea16d71c0a80ee76010bd4ce</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WF...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENCFF000WFM</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EUL</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>3236ffaaff5e5c719444d8c107ca5c77</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WF...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENCFF000ARG</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKB</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>881c311f387465059bace19e517bf47c</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AR...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENCFF000ARI</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKB</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>d53da79ce1f1927b77a938288b758180</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AR...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENCFF000NSJ</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BGT</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>0e6303469c9cb86a61f1814c21c69cb9</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000NS...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENCFF000NSN</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BGT</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>003671b1058149cc2c347dd76cf0d507</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000NS...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENCFF000OGJ</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BQT</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>0f0cae2bba5e566c9835e745abc080c3</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000OG...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENCFF000OGM</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BQT</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>f6c55e80781d9867d918c720217810f6</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000OG...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENCFF000ROM</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DKV</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>49a0bd6730c3950087edd7162a7a286d</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000RO...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENCFF000RON</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DKV</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>24c25b458cd7e472af7cff89b91ce706</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000RO...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENCFF000ROQ</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DKV</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>c7afb31b2e0b6a6c11c3f7f6793ff9cc</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000RO...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENCFF000VTR</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DZR</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>3a30d22938d5e825afa7ec25258c09cf</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000VT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENCFF000VTS</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DZR</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>910c252082a3fd517f0ffeb0a36d80ea</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000VT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENCFF000WBH</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>0a3746183a82deed645674cec5dbbb37</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENCFF000WBI</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>d202ab81b94d9e1efca9e928977f87f6</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENCFF000WBJ</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>ae39aad4ef08e51aba5fc84691de309b</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENCFF000WBK</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>938a196875dabe07694e6eaef308f0cc</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENCFF000WBL</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>eb74d7f271e9aed6c8f2f316900a9e71</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENCFF000WBN</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>d6ec5fcc7669935c34f522e2f6342f49</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENCFF000WBO</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000EAD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>73a091e4c66231fba696810399acb1e5</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WB...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENCFF000WDT</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DYU</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>f9b8f860050d7ef136b965e8fc012d8e</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WD...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENCFF000WDU</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DYU</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>c39d026208ecc27912d90fbe3cf822ca</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000WD...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENCFF000VUC</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DZN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>e735ec94ce4d7716373aa0a3e1769e6b</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000VU...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENCFF000VUE</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000DZN</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>f4dd7412a968abb7f5e62eee1a80113d</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000VU...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENCFF000NTF</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BJZ</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>6e3a790e78b9858d207af4d0f2481a49</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000NT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENCFF000NTG</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BJZ</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>79106e1fe92f00b8ec9582057afc6137</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000NT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENCFF000NYZ</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BHD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>7aeca89582e6782a0d92691fceae94f6</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000NY...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENCFF000NZF</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000BHD</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>e2ef2288343a6c53b9a038a9d0dc92d8</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000NZ...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biological replicates with identical biosample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>ENCFF231IAK</td>\n",
       "      <td>bam</td>\n",
       "      <td>unfiltered alignments</td>\n",
       "      <td>ENCSR000AKJ</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>ca8f87f07ffdfcaeea0be85d60e45cdb</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF231IA...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mismatched file status, biological replicates ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>ENCFF651UIO</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKJ</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>4effce3d5e505846deba3998eed76963</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF651UI...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mismatched file status, biological replicates ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>ENCFF858ORL</td>\n",
       "      <td>bam</td>\n",
       "      <td>unfiltered alignments</td>\n",
       "      <td>ENCSR000AKJ</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>f81b6321dcab0816c453dd03e2688f45</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF858OR...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mismatched file status, biological replicates ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>ENCFF684NLR</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKJ</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>bcfe0e347d0bae688191fc3e48cfacb7</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF684NL...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mismatched file status, biological replicates ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>ENCFF000ASG</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKC</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>aeb9ded9b366bc49038af4d450581099</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AS...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>ENCFF000ASI</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKC</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>89ce74342e28239b732aa7639044509b</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AS...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>ENCFF620IUQ</td>\n",
       "      <td>bam</td>\n",
       "      <td>unfiltered alignments</td>\n",
       "      <td>ENCSR000AKC</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>bd3f3db6d3062037dc8a0befcd5d7aaf</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF620IU...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>ENCFF948GTC</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKC</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>7c682741667cff3052480e0ec1cb801a</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF948GT...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>control insufficient read depth</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>828</th>\n",
       "      <td>ENCFF067FBS</td>\n",
       "      <td>bam</td>\n",
       "      <td>unfiltered alignments</td>\n",
       "      <td>ENCSR000AKC</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>5607b845f8d623b6d87f9ae06d8056fd</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF067FB...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>829</th>\n",
       "      <td>ENCFF804NCH</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKC</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>02718c8c66af7b5074128ff694bbdec6</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF804NC...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>control insufficient read depth</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>ENCFF000ATT</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AOW</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>f084b978269665a5e41db6e947e4edc5</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>ENCFF000ATV</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AOW</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>0b4dedd28987807e756aafe5755a1350</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>ENCFF000ARR</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AOV</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>682fb3ab3a3721f8f6589791159f119e</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AR...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>ENCFF000ARS</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AOV</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>371bb53d4468fed58efe0af367c344ff</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AR...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>ENCFF000ARY</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKF</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>643b2e292c47652540947eb980a2d434</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AR...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>ENCFF000ATE</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKF</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>dfa727df060e2ecf2d05703a95b5152f</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AT...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>ENCFF000AUR</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKI</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>a41aab03b2f6e3472e3e20d99b79b210</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AU...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>837</th>\n",
       "      <td>ENCFF000AUT</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000AKI</td>\n",
       "      <td>ChIP-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>2aec140d26a96cda9c79ddc4bbc518e2</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000AU...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>mixed read lengths</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>ENCFF000EVY</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>ef50cfac8b54beae2e40b510516dbfbe</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000EV...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>ENCFF000EVZ</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>d5c629b8f777487a31f8b47d0851675e</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000EV...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>840</th>\n",
       "      <td>ENCFF595AWE</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>bf7d127b77c4c748190b9f3358780a7c</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF595AW...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>ENCFF148TMK</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>1ce6f0f31e72ca694894acaeaf1daab2</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF148TM...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>archived</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>ENCFF709IUX</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>6c1f818cc0e7eb01977c02efa867f75d</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF709IU...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>843</th>\n",
       "      <td>ENCFF235OBA</td>\n",
       "      <td>bam</td>\n",
       "      <td>transcriptome alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>8e49aa469b768a4450ad401f8e123ef2</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF235OB...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>844</th>\n",
       "      <td>ENCFF244ZQA</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>db8cdfee77a8105c282984d2e4f7fc9a</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF244ZQ...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>845</th>\n",
       "      <td>ENCFF923SVC</td>\n",
       "      <td>bam</td>\n",
       "      <td>transcriptome alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>83353cc5a718b03c4b785a4ee299190d</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF923SV...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>846</th>\n",
       "      <td>ENCFF664VNB</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>5bd6ad10895a190a0c0559ec113b33fe</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF664VN...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>847</th>\n",
       "      <td>ENCFF271YOW</td>\n",
       "      <td>bam</td>\n",
       "      <td>transcriptome alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>a2721f81424fddd7e5c3db877d3cd764</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF271YO...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>ENCFF790WER</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>402b1997459e342965ca0d4d2284f86b</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF790WE...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>849</th>\n",
       "      <td>ENCFF862SIX</td>\n",
       "      <td>bam</td>\n",
       "      <td>transcriptome alignments</td>\n",
       "      <td>ENCSR000COQ</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>5d0214a7d869e1356ac436b344755b05</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF862SI...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>850 rows  48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    File accession File format               Output type Experiment accession  \\\n",
       "0      ENCFF000IZE         bam                alignments          ENCSR000CQV   \n",
       "1      ENCFF000IZF         bam                alignments          ENCSR000CQV   \n",
       "2      ENCFF000WFL         bam                alignments          ENCSR000EUL   \n",
       "3      ENCFF000WFM         bam                alignments          ENCSR000EUL   \n",
       "4      ENCFF000ARG         bam                alignments          ENCSR000AKB   \n",
       "5      ENCFF000ARI         bam                alignments          ENCSR000AKB   \n",
       "6      ENCFF000NSJ         bam                alignments          ENCSR000BGT   \n",
       "7      ENCFF000NSN         bam                alignments          ENCSR000BGT   \n",
       "8      ENCFF000OGJ         bam                alignments          ENCSR000BQT   \n",
       "9      ENCFF000OGM         bam                alignments          ENCSR000BQT   \n",
       "10     ENCFF000ROM         bam                alignments          ENCSR000DKV   \n",
       "11     ENCFF000RON         bam                alignments          ENCSR000DKV   \n",
       "12     ENCFF000ROQ         bam                alignments          ENCSR000DKV   \n",
       "13     ENCFF000VTR         bam                alignments          ENCSR000DZR   \n",
       "14     ENCFF000VTS         bam                alignments          ENCSR000DZR   \n",
       "15     ENCFF000WBH         bam                alignments          ENCSR000EAD   \n",
       "16     ENCFF000WBI         bam                alignments          ENCSR000EAD   \n",
       "17     ENCFF000WBJ         bam                alignments          ENCSR000EAD   \n",
       "18     ENCFF000WBK         bam                alignments          ENCSR000EAD   \n",
       "19     ENCFF000WBL         bam                alignments          ENCSR000EAD   \n",
       "20     ENCFF000WBN         bam                alignments          ENCSR000EAD   \n",
       "21     ENCFF000WBO         bam                alignments          ENCSR000EAD   \n",
       "22     ENCFF000WDT         bam                alignments          ENCSR000DYU   \n",
       "23     ENCFF000WDU         bam                alignments          ENCSR000DYU   \n",
       "24     ENCFF000VUC         bam                alignments          ENCSR000DZN   \n",
       "25     ENCFF000VUE         bam                alignments          ENCSR000DZN   \n",
       "26     ENCFF000NTF         bam                alignments          ENCSR000BJZ   \n",
       "27     ENCFF000NTG         bam                alignments          ENCSR000BJZ   \n",
       "28     ENCFF000NYZ         bam                alignments          ENCSR000BHD   \n",
       "29     ENCFF000NZF         bam                alignments          ENCSR000BHD   \n",
       "..             ...         ...                       ...                  ...   \n",
       "820    ENCFF231IAK         bam     unfiltered alignments          ENCSR000AKJ   \n",
       "821    ENCFF651UIO         bam                alignments          ENCSR000AKJ   \n",
       "822    ENCFF858ORL         bam     unfiltered alignments          ENCSR000AKJ   \n",
       "823    ENCFF684NLR         bam                alignments          ENCSR000AKJ   \n",
       "824    ENCFF000ASG         bam                alignments          ENCSR000AKC   \n",
       "825    ENCFF000ASI         bam                alignments          ENCSR000AKC   \n",
       "826    ENCFF620IUQ         bam     unfiltered alignments          ENCSR000AKC   \n",
       "827    ENCFF948GTC         bam                alignments          ENCSR000AKC   \n",
       "828    ENCFF067FBS         bam     unfiltered alignments          ENCSR000AKC   \n",
       "829    ENCFF804NCH         bam                alignments          ENCSR000AKC   \n",
       "830    ENCFF000ATT         bam                alignments          ENCSR000AOW   \n",
       "831    ENCFF000ATV         bam                alignments          ENCSR000AOW   \n",
       "832    ENCFF000ARR         bam                alignments          ENCSR000AOV   \n",
       "833    ENCFF000ARS         bam                alignments          ENCSR000AOV   \n",
       "834    ENCFF000ARY         bam                alignments          ENCSR000AKF   \n",
       "835    ENCFF000ATE         bam                alignments          ENCSR000AKF   \n",
       "836    ENCFF000AUR         bam                alignments          ENCSR000AKI   \n",
       "837    ENCFF000AUT         bam                alignments          ENCSR000AKI   \n",
       "838    ENCFF000EVY         bam                alignments          ENCSR000COQ   \n",
       "839    ENCFF000EVZ         bam                alignments          ENCSR000COQ   \n",
       "840    ENCFF595AWE         bam                alignments          ENCSR000COQ   \n",
       "841    ENCFF148TMK         bam                alignments          ENCSR000COQ   \n",
       "842    ENCFF709IUX         bam                alignments          ENCSR000COQ   \n",
       "843    ENCFF235OBA         bam  transcriptome alignments          ENCSR000COQ   \n",
       "844    ENCFF244ZQA         bam                alignments          ENCSR000COQ   \n",
       "845    ENCFF923SVC         bam  transcriptome alignments          ENCSR000COQ   \n",
       "846    ENCFF664VNB         bam                alignments          ENCSR000COQ   \n",
       "847    ENCFF271YOW         bam  transcriptome alignments          ENCSR000COQ   \n",
       "848    ENCFF790WER         bam                alignments          ENCSR000COQ   \n",
       "849    ENCFF862SIX         bam  transcriptome alignments          ENCSR000COQ   \n",
       "\n",
       "        Assay Biosample term id Biosample term name          Biosample type  \\\n",
       "0     RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "1     RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "2    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "3    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "4    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "5    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "6    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "7    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "8    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "9    ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "10   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "11   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "12   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "13   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "14   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "15   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "16   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "17   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "18   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "19   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "20   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "21   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "22   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "23   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "24   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "25   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "26   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "27   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "28   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "29   ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "..        ...               ...                 ...                     ...   \n",
       "820  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "821  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "822  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "823  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "824  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "825  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "826  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "827  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "828  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "829  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "830  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "831  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "832  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "833  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "834  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "835  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "836  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "837  ChIP-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "838   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "839   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "840   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "841   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "842   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "843   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "844   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "845   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "846   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "847   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "848   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "849   RNA-seq       EFO:0002784             GM12878  immortalized cell line   \n",
       "\n",
       "    Biosample life stage Biosample sex     ...      \\\n",
       "0                  adult        female     ...       \n",
       "1                  adult        female     ...       \n",
       "2                  adult        female     ...       \n",
       "3                  adult        female     ...       \n",
       "4                  adult        female     ...       \n",
       "5                  adult        female     ...       \n",
       "6                  adult        female     ...       \n",
       "7                  adult        female     ...       \n",
       "8                  adult        female     ...       \n",
       "9                  adult        female     ...       \n",
       "10                 adult        female     ...       \n",
       "11                 adult        female     ...       \n",
       "12                 adult        female     ...       \n",
       "13                 adult        female     ...       \n",
       "14                 adult        female     ...       \n",
       "15                 adult        female     ...       \n",
       "16                 adult        female     ...       \n",
       "17                 adult        female     ...       \n",
       "18                 adult        female     ...       \n",
       "19                 adult        female     ...       \n",
       "20                 adult        female     ...       \n",
       "21                 adult        female     ...       \n",
       "22                 adult        female     ...       \n",
       "23                 adult        female     ...       \n",
       "24                 adult        female     ...       \n",
       "25                 adult        female     ...       \n",
       "26                 adult        female     ...       \n",
       "27                 adult        female     ...       \n",
       "28                 adult        female     ...       \n",
       "29                 adult        female     ...       \n",
       "..                   ...           ...     ...       \n",
       "820                adult        female     ...       \n",
       "821                adult        female     ...       \n",
       "822                adult        female     ...       \n",
       "823                adult        female     ...       \n",
       "824                adult        female     ...       \n",
       "825                adult        female     ...       \n",
       "826                adult        female     ...       \n",
       "827                adult        female     ...       \n",
       "828                adult        female     ...       \n",
       "829                adult        female     ...       \n",
       "830                adult        female     ...       \n",
       "831                adult        female     ...       \n",
       "832                adult        female     ...       \n",
       "833                adult        female     ...       \n",
       "834                adult        female     ...       \n",
       "835                adult        female     ...       \n",
       "836                adult        female     ...       \n",
       "837                adult        female     ...       \n",
       "838                adult        female     ...       \n",
       "839                adult        female     ...       \n",
       "840                adult        female     ...       \n",
       "841                adult        female     ...       \n",
       "842                adult        female     ...       \n",
       "843                adult        female     ...       \n",
       "844                adult        female     ...       \n",
       "845                adult        female     ...       \n",
       "846                adult        female     ...       \n",
       "847                adult        female     ...       \n",
       "848                adult        female     ...       \n",
       "849                adult        female     ...       \n",
       "\n",
       "                               md5sum  \\\n",
       "0    32e18ca6acd794dbbb9ce8cc1f986f10   \n",
       "1    e41f8dd0aaf9c019f6f1f055c8ea9952   \n",
       "2    b91dcf11ea16d71c0a80ee76010bd4ce   \n",
       "3    3236ffaaff5e5c719444d8c107ca5c77   \n",
       "4    881c311f387465059bace19e517bf47c   \n",
       "5    d53da79ce1f1927b77a938288b758180   \n",
       "6    0e6303469c9cb86a61f1814c21c69cb9   \n",
       "7    003671b1058149cc2c347dd76cf0d507   \n",
       "8    0f0cae2bba5e566c9835e745abc080c3   \n",
       "9    f6c55e80781d9867d918c720217810f6   \n",
       "10   49a0bd6730c3950087edd7162a7a286d   \n",
       "11   24c25b458cd7e472af7cff89b91ce706   \n",
       "12   c7afb31b2e0b6a6c11c3f7f6793ff9cc   \n",
       "13   3a30d22938d5e825afa7ec25258c09cf   \n",
       "14   910c252082a3fd517f0ffeb0a36d80ea   \n",
       "15   0a3746183a82deed645674cec5dbbb37   \n",
       "16   d202ab81b94d9e1efca9e928977f87f6   \n",
       "17   ae39aad4ef08e51aba5fc84691de309b   \n",
       "18   938a196875dabe07694e6eaef308f0cc   \n",
       "19   eb74d7f271e9aed6c8f2f316900a9e71   \n",
       "20   d6ec5fcc7669935c34f522e2f6342f49   \n",
       "21   73a091e4c66231fba696810399acb1e5   \n",
       "22   f9b8f860050d7ef136b965e8fc012d8e   \n",
       "23   c39d026208ecc27912d90fbe3cf822ca   \n",
       "24   e735ec94ce4d7716373aa0a3e1769e6b   \n",
       "25   f4dd7412a968abb7f5e62eee1a80113d   \n",
       "26   6e3a790e78b9858d207af4d0f2481a49   \n",
       "27   79106e1fe92f00b8ec9582057afc6137   \n",
       "28   7aeca89582e6782a0d92691fceae94f6   \n",
       "29   e2ef2288343a6c53b9a038a9d0dc92d8   \n",
       "..                                ...   \n",
       "820  ca8f87f07ffdfcaeea0be85d60e45cdb   \n",
       "821  4effce3d5e505846deba3998eed76963   \n",
       "822  f81b6321dcab0816c453dd03e2688f45   \n",
       "823  bcfe0e347d0bae688191fc3e48cfacb7   \n",
       "824  aeb9ded9b366bc49038af4d450581099   \n",
       "825  89ce74342e28239b732aa7639044509b   \n",
       "826  bd3f3db6d3062037dc8a0befcd5d7aaf   \n",
       "827  7c682741667cff3052480e0ec1cb801a   \n",
       "828  5607b845f8d623b6d87f9ae06d8056fd   \n",
       "829  02718c8c66af7b5074128ff694bbdec6   \n",
       "830  f084b978269665a5e41db6e947e4edc5   \n",
       "831  0b4dedd28987807e756aafe5755a1350   \n",
       "832  682fb3ab3a3721f8f6589791159f119e   \n",
       "833  371bb53d4468fed58efe0af367c344ff   \n",
       "834  643b2e292c47652540947eb980a2d434   \n",
       "835  dfa727df060e2ecf2d05703a95b5152f   \n",
       "836  a41aab03b2f6e3472e3e20d99b79b210   \n",
       "837  2aec140d26a96cda9c79ddc4bbc518e2   \n",
       "838  ef50cfac8b54beae2e40b510516dbfbe   \n",
       "839  d5c629b8f777487a31f8b47d0851675e   \n",
       "840  bf7d127b77c4c748190b9f3358780a7c   \n",
       "841  1ce6f0f31e72ca694894acaeaf1daab2   \n",
       "842  6c1f818cc0e7eb01977c02efa867f75d   \n",
       "843  8e49aa469b768a4450ad401f8e123ef2   \n",
       "844  db8cdfee77a8105c282984d2e4f7fc9a   \n",
       "845  83353cc5a718b03c4b785a4ee299190d   \n",
       "846  5bd6ad10895a190a0c0559ec113b33fe   \n",
       "847  a2721f81424fddd7e5c3db877d3cd764   \n",
       "848  402b1997459e342965ca0d4d2284f86b   \n",
       "849  5d0214a7d869e1356ac436b344755b05   \n",
       "\n",
       "                                     File download URL Assembly Platform  \\\n",
       "0    https://www.encodeproject.org/files/ENCFF000IZ...     hg19      NaN   \n",
       "1    https://www.encodeproject.org/files/ENCFF000IZ...     hg19      NaN   \n",
       "2    https://www.encodeproject.org/files/ENCFF000WF...     hg19      NaN   \n",
       "3    https://www.encodeproject.org/files/ENCFF000WF...     hg19      NaN   \n",
       "4    https://www.encodeproject.org/files/ENCFF000AR...     hg19      NaN   \n",
       "5    https://www.encodeproject.org/files/ENCFF000AR...     hg19      NaN   \n",
       "6    https://www.encodeproject.org/files/ENCFF000NS...     hg19      NaN   \n",
       "7    https://www.encodeproject.org/files/ENCFF000NS...     hg19      NaN   \n",
       "8    https://www.encodeproject.org/files/ENCFF000OG...     hg19      NaN   \n",
       "9    https://www.encodeproject.org/files/ENCFF000OG...     hg19      NaN   \n",
       "10   https://www.encodeproject.org/files/ENCFF000RO...     hg19      NaN   \n",
       "11   https://www.encodeproject.org/files/ENCFF000RO...     hg19      NaN   \n",
       "12   https://www.encodeproject.org/files/ENCFF000RO...     hg19      NaN   \n",
       "13   https://www.encodeproject.org/files/ENCFF000VT...     hg19      NaN   \n",
       "14   https://www.encodeproject.org/files/ENCFF000VT...     hg19      NaN   \n",
       "15   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "16   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "17   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "18   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "19   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "20   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "21   https://www.encodeproject.org/files/ENCFF000WB...     hg19      NaN   \n",
       "22   https://www.encodeproject.org/files/ENCFF000WD...     hg19      NaN   \n",
       "23   https://www.encodeproject.org/files/ENCFF000WD...     hg19      NaN   \n",
       "24   https://www.encodeproject.org/files/ENCFF000VU...     hg19      NaN   \n",
       "25   https://www.encodeproject.org/files/ENCFF000VU...     hg19      NaN   \n",
       "26   https://www.encodeproject.org/files/ENCFF000NT...     hg19      NaN   \n",
       "27   https://www.encodeproject.org/files/ENCFF000NT...     hg19      NaN   \n",
       "28   https://www.encodeproject.org/files/ENCFF000NY...     hg19      NaN   \n",
       "29   https://www.encodeproject.org/files/ENCFF000NZ...     hg19      NaN   \n",
       "..                                                 ...      ...      ...   \n",
       "820  https://www.encodeproject.org/files/ENCFF231IA...   GRCh38      NaN   \n",
       "821  https://www.encodeproject.org/files/ENCFF651UI...   GRCh38      NaN   \n",
       "822  https://www.encodeproject.org/files/ENCFF858OR...   GRCh38      NaN   \n",
       "823  https://www.encodeproject.org/files/ENCFF684NL...   GRCh38      NaN   \n",
       "824  https://www.encodeproject.org/files/ENCFF000AS...     hg19      NaN   \n",
       "825  https://www.encodeproject.org/files/ENCFF000AS...     hg19      NaN   \n",
       "826  https://www.encodeproject.org/files/ENCFF620IU...   GRCh38      NaN   \n",
       "827  https://www.encodeproject.org/files/ENCFF948GT...   GRCh38      NaN   \n",
       "828  https://www.encodeproject.org/files/ENCFF067FB...   GRCh38      NaN   \n",
       "829  https://www.encodeproject.org/files/ENCFF804NC...   GRCh38      NaN   \n",
       "830  https://www.encodeproject.org/files/ENCFF000AT...     hg19      NaN   \n",
       "831  https://www.encodeproject.org/files/ENCFF000AT...     hg19      NaN   \n",
       "832  https://www.encodeproject.org/files/ENCFF000AR...     hg19      NaN   \n",
       "833  https://www.encodeproject.org/files/ENCFF000AR...     hg19      NaN   \n",
       "834  https://www.encodeproject.org/files/ENCFF000AR...     hg19      NaN   \n",
       "835  https://www.encodeproject.org/files/ENCFF000AT...     hg19      NaN   \n",
       "836  https://www.encodeproject.org/files/ENCFF000AU...     hg19      NaN   \n",
       "837  https://www.encodeproject.org/files/ENCFF000AU...     hg19      NaN   \n",
       "838  https://www.encodeproject.org/files/ENCFF000EV...     hg19      NaN   \n",
       "839  https://www.encodeproject.org/files/ENCFF000EV...     hg19      NaN   \n",
       "840  https://www.encodeproject.org/files/ENCFF595AW...   GRCh38      NaN   \n",
       "841  https://www.encodeproject.org/files/ENCFF148TM...   GRCh38      NaN   \n",
       "842  https://www.encodeproject.org/files/ENCFF709IU...   GRCh38      NaN   \n",
       "843  https://www.encodeproject.org/files/ENCFF235OB...   GRCh38      NaN   \n",
       "844  https://www.encodeproject.org/files/ENCFF244ZQ...   GRCh38      NaN   \n",
       "845  https://www.encodeproject.org/files/ENCFF923SV...   GRCh38      NaN   \n",
       "846  https://www.encodeproject.org/files/ENCFF664VN...     hg19      NaN   \n",
       "847  https://www.encodeproject.org/files/ENCFF271YO...     hg19      NaN   \n",
       "848  https://www.encodeproject.org/files/ENCFF790WE...     hg19      NaN   \n",
       "849  https://www.encodeproject.org/files/ENCFF862SI...     hg19      NaN   \n",
       "\n",
       "     Controlled by File Status       Audit WARNING  \\\n",
       "0              NaN    released                 NaN   \n",
       "1              NaN    released                 NaN   \n",
       "2              NaN    released                 NaN   \n",
       "3              NaN    released                 NaN   \n",
       "4              NaN    released  mixed read lengths   \n",
       "5              NaN    released  mixed read lengths   \n",
       "6              NaN    released                 NaN   \n",
       "7              NaN    released                 NaN   \n",
       "8              NaN    released                 NaN   \n",
       "9              NaN    released                 NaN   \n",
       "10             NaN    released                 NaN   \n",
       "11             NaN    released                 NaN   \n",
       "12             NaN    released                 NaN   \n",
       "13             NaN    released                 NaN   \n",
       "14             NaN    released                 NaN   \n",
       "15             NaN    released  mixed read lengths   \n",
       "16             NaN    released  mixed read lengths   \n",
       "17             NaN    released  mixed read lengths   \n",
       "18             NaN    released  mixed read lengths   \n",
       "19             NaN    released  mixed read lengths   \n",
       "20             NaN    released  mixed read lengths   \n",
       "21             NaN    released  mixed read lengths   \n",
       "22             NaN    released                 NaN   \n",
       "23             NaN    released                 NaN   \n",
       "24             NaN    released                 NaN   \n",
       "25             NaN    released                 NaN   \n",
       "26             NaN    released                 NaN   \n",
       "27             NaN    released                 NaN   \n",
       "28             NaN    released                 NaN   \n",
       "29             NaN    released                 NaN   \n",
       "..             ...         ...                 ...   \n",
       "820            NaN    released                 NaN   \n",
       "821            NaN    released                 NaN   \n",
       "822            NaN    released                 NaN   \n",
       "823            NaN    released                 NaN   \n",
       "824            NaN    released                 NaN   \n",
       "825            NaN    released                 NaN   \n",
       "826            NaN    released                 NaN   \n",
       "827            NaN    released                 NaN   \n",
       "828            NaN    released                 NaN   \n",
       "829            NaN    released                 NaN   \n",
       "830            NaN    released                 NaN   \n",
       "831            NaN    released                 NaN   \n",
       "832            NaN    released                 NaN   \n",
       "833            NaN    released                 NaN   \n",
       "834            NaN    released  mixed read lengths   \n",
       "835            NaN    released  mixed read lengths   \n",
       "836            NaN    released  mixed read lengths   \n",
       "837            NaN    released  mixed read lengths   \n",
       "838            NaN    archived                 NaN   \n",
       "839            NaN    archived                 NaN   \n",
       "840            NaN    archived                 NaN   \n",
       "841            NaN    archived                 NaN   \n",
       "842            NaN    released                 NaN   \n",
       "843            NaN    released                 NaN   \n",
       "844            NaN    released                 NaN   \n",
       "845            NaN    released                 NaN   \n",
       "846            NaN    released                 NaN   \n",
       "847            NaN    released                 NaN   \n",
       "848            NaN    released                 NaN   \n",
       "849            NaN    released                 NaN   \n",
       "\n",
       "                                 Audit INTERNAL_ACTION  \\\n",
       "0                                                  NaN   \n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3                                                  NaN   \n",
       "4                                                  NaN   \n",
       "5                                                  NaN   \n",
       "6       biological replicates with identical biosample   \n",
       "7       biological replicates with identical biosample   \n",
       "8       biological replicates with identical biosample   \n",
       "9       biological replicates with identical biosample   \n",
       "10      biological replicates with identical biosample   \n",
       "11      biological replicates with identical biosample   \n",
       "12      biological replicates with identical biosample   \n",
       "13      biological replicates with identical biosample   \n",
       "14      biological replicates with identical biosample   \n",
       "15      biological replicates with identical biosample   \n",
       "16      biological replicates with identical biosample   \n",
       "17      biological replicates with identical biosample   \n",
       "18      biological replicates with identical biosample   \n",
       "19      biological replicates with identical biosample   \n",
       "20      biological replicates with identical biosample   \n",
       "21      biological replicates with identical biosample   \n",
       "22      biological replicates with identical biosample   \n",
       "23      biological replicates with identical biosample   \n",
       "24      biological replicates with identical biosample   \n",
       "25      biological replicates with identical biosample   \n",
       "26      biological replicates with identical biosample   \n",
       "27      biological replicates with identical biosample   \n",
       "28      biological replicates with identical biosample   \n",
       "29      biological replicates with identical biosample   \n",
       "..                                                 ...   \n",
       "820  mismatched file status, biological replicates ...   \n",
       "821  mismatched file status, biological replicates ...   \n",
       "822  mismatched file status, biological replicates ...   \n",
       "823  mismatched file status, biological replicates ...   \n",
       "824                                                NaN   \n",
       "825                                                NaN   \n",
       "826                                                NaN   \n",
       "827                                                NaN   \n",
       "828                                                NaN   \n",
       "829                                                NaN   \n",
       "830                                                NaN   \n",
       "831                                                NaN   \n",
       "832                                                NaN   \n",
       "833                                                NaN   \n",
       "834                                                NaN   \n",
       "835                                                NaN   \n",
       "836                                                NaN   \n",
       "837                                                NaN   \n",
       "838                                                NaN   \n",
       "839                                                NaN   \n",
       "840                                                NaN   \n",
       "841                                                NaN   \n",
       "842                                                NaN   \n",
       "843                                                NaN   \n",
       "844                                                NaN   \n",
       "845                                                NaN   \n",
       "846                                                NaN   \n",
       "847                                                NaN   \n",
       "848                                                NaN   \n",
       "849                                                NaN   \n",
       "\n",
       "                 Audit NOT_COMPLIANT Audit ERROR  \n",
       "0                                NaN         NaN  \n",
       "1                                NaN         NaN  \n",
       "2                                NaN         NaN  \n",
       "3                                NaN         NaN  \n",
       "4                                NaN         NaN  \n",
       "5                                NaN         NaN  \n",
       "6                                NaN         NaN  \n",
       "7                                NaN         NaN  \n",
       "8                                NaN         NaN  \n",
       "9                                NaN         NaN  \n",
       "10                               NaN         NaN  \n",
       "11                               NaN         NaN  \n",
       "12                               NaN         NaN  \n",
       "13                               NaN         NaN  \n",
       "14                               NaN         NaN  \n",
       "15                               NaN         NaN  \n",
       "16                               NaN         NaN  \n",
       "17                               NaN         NaN  \n",
       "18                               NaN         NaN  \n",
       "19                               NaN         NaN  \n",
       "20                               NaN         NaN  \n",
       "21                               NaN         NaN  \n",
       "22                               NaN         NaN  \n",
       "23                               NaN         NaN  \n",
       "24                               NaN         NaN  \n",
       "25                               NaN         NaN  \n",
       "26                               NaN         NaN  \n",
       "27                               NaN         NaN  \n",
       "28                               NaN         NaN  \n",
       "29                               NaN         NaN  \n",
       "..                               ...         ...  \n",
       "820                              NaN         NaN  \n",
       "821                              NaN         NaN  \n",
       "822                              NaN         NaN  \n",
       "823                              NaN         NaN  \n",
       "824                              NaN         NaN  \n",
       "825                              NaN         NaN  \n",
       "826                              NaN         NaN  \n",
       "827  control insufficient read depth         NaN  \n",
       "828                              NaN         NaN  \n",
       "829  control insufficient read depth         NaN  \n",
       "830                              NaN         NaN  \n",
       "831                              NaN         NaN  \n",
       "832                              NaN         NaN  \n",
       "833                              NaN         NaN  \n",
       "834                              NaN         NaN  \n",
       "835                              NaN         NaN  \n",
       "836                              NaN         NaN  \n",
       "837                              NaN         NaN  \n",
       "838                              NaN         NaN  \n",
       "839                              NaN         NaN  \n",
       "840                              NaN         NaN  \n",
       "841                              NaN         NaN  \n",
       "842                              NaN         NaN  \n",
       "843                              NaN         NaN  \n",
       "844                              NaN         NaN  \n",
       "845                              NaN         NaN  \n",
       "846                              NaN         NaN  \n",
       "847                              NaN         NaN  \n",
       "848                              NaN         NaN  \n",
       "849                              NaN         NaN  \n",
       "\n",
       "[850 rows x 48 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table(jp('metadata.tsv'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>File accession</th>\n",
       "      <th>File format</th>\n",
       "      <th>Output type</th>\n",
       "      <th>Experiment accession</th>\n",
       "      <th>Assay</th>\n",
       "      <th>Biosample term id</th>\n",
       "      <th>Biosample term name</th>\n",
       "      <th>Biosample type</th>\n",
       "      <th>Biosample life stage</th>\n",
       "      <th>Biosample sex</th>\n",
       "      <th>...</th>\n",
       "      <th>md5sum</th>\n",
       "      <th>File download URL</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Platform</th>\n",
       "      <th>Controlled by</th>\n",
       "      <th>File Status</th>\n",
       "      <th>Audit WARNING</th>\n",
       "      <th>Audit INTERNAL_ACTION</th>\n",
       "      <th>Audit NOT_COMPLIANT</th>\n",
       "      <th>Audit ERROR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENCFF000IZE</td>\n",
       "      <td>bam</td>\n",
       "      <td>alignments</td>\n",
       "      <td>ENCSR000CQV</td>\n",
       "      <td>RNA-seq</td>\n",
       "      <td>EFO:0002784</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>immortalized cell line</td>\n",
       "      <td>adult</td>\n",
       "      <td>female</td>\n",
       "      <td>...</td>\n",
       "      <td>32e18ca6acd794dbbb9ce8cc1f986f10</td>\n",
       "      <td>https://www.encodeproject.org/files/ENCFF000IZ...</td>\n",
       "      <td>hg19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>released</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows  48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  File accession File format Output type Experiment accession    Assay  \\\n",
       "0    ENCFF000IZE         bam  alignments          ENCSR000CQV  RNA-seq   \n",
       "\n",
       "  Biosample term id Biosample term name          Biosample type  \\\n",
       "0       EFO:0002784             GM12878  immortalized cell line   \n",
       "\n",
       "  Biosample life stage Biosample sex     ...      \\\n",
       "0                adult        female     ...       \n",
       "\n",
       "                             md5sum  \\\n",
       "0  32e18ca6acd794dbbb9ce8cc1f986f10   \n",
       "\n",
       "                                   File download URL Assembly Platform  \\\n",
       "0  https://www.encodeproject.org/files/ENCFF000IZ...     hg19      NaN   \n",
       "\n",
       "   Controlled by File Status Audit WARNING Audit INTERNAL_ACTION  \\\n",
       "0            NaN    released           NaN                   NaN   \n",
       "\n",
       "  Audit NOT_COMPLIANT Audit ERROR  \n",
       "0                 NaN         NaN  \n",
       "\n",
       "[1 rows x 48 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata=pd.read_table(jp('metadata.tsv'))\n",
    "metadata[:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'File accession', u'File format', u'Output type',\n",
       "       u'Experiment accession', u'Assay', u'Biosample term id',\n",
       "       u'Biosample term name', u'Biosample type', u'Biosample life stage',\n",
       "       u'Biosample sex', u'Biosample organism', u'Biosample treatments',\n",
       "       u'Biosample subcellular fraction term name', u'Biosample phase',\n",
       "       u'Biosample synchronization stage', u'Experiment target',\n",
       "       u'Antibody accession', u'Library made from', u'Library depleted in',\n",
       "       u'Library extraction method', u'Library lysis method',\n",
       "       u'Library crosslinking method', u'Experiment date released', u'Project',\n",
       "       u'RBNS protein concentration', u'Library fragmentation method',\n",
       "       u'Library size range', u'Biosample Age', u'Biological replicate(s)',\n",
       "       u'Technical replicate', u'Read length', u'Mapped read length',\n",
       "       u'Run type', u'Paired end', u'Paired with', u'Derived from', u'Size',\n",
       "       u'Lab', u'md5sum', u'File download URL', u'Assembly', u'Platform',\n",
       "       u'Controlled by', u'File Status', u'Audit WARNING',\n",
       "       u'Audit INTERNAL_ACTION', u'Audit NOT_COMPLIANT', u'Audit ERROR'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>average_coverage</th>\n",
       "      <th>20_total_cvg</th>\n",
       "      <th>20_mean_cvg</th>\n",
       "      <th>20_granular_Q1</th>\n",
       "      <th>20_granular_median</th>\n",
       "      <th>20_granular_Q3</th>\n",
       "      <th>20_%_above_15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>POTEH</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OR11H1</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCT8L2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>XKR3</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GAB4</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>IL17RA</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CECR6</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CECR5</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CECR1</td>\n",
       "      <td>23291502</td>\n",
       "      <td>0.08</td>\n",
       "      <td>23291502</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CECR2</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SLC25A18</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LOC101929372</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ATP6V1E1</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>BCL2L13</td>\n",
       "      <td>42701087</td>\n",
       "      <td>0.08</td>\n",
       "      <td>42701087</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>BID</td>\n",
       "      <td>27173419</td>\n",
       "      <td>0.08</td>\n",
       "      <td>27173419</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MICAL3</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PEX26</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TUBA8</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>USP18</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DGCR6</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PRODH</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>DGCR2</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>DGCR14</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TSSK2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GSC2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SLC25A1</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CLTCL1</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>HIRA</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MRPL40</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>C22orf39</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>IL17REL</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>400</th>\n",
       "      <td>MLC1</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>MOV10L1</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>PANX2</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>403</th>\n",
       "      <td>TRABD</td>\n",
       "      <td>19409585</td>\n",
       "      <td>0.08</td>\n",
       "      <td>19409585</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>SELENOO</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>TUBGCP6</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>HDAC10</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>MAPK12</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7763834</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>MAPK11</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>PLXNB2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>DENND6B</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>PPP6R2</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>SBF1</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>413</th>\n",
       "      <td>ADM2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>MIOX</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>LMF2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>NCAPH2</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>11645751</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>SCO2</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>15527668</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>TYMP</td>\n",
       "      <td>19409585</td>\n",
       "      <td>0.08</td>\n",
       "      <td>19409585</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>ODF3B</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>KLHDC7B</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>SYCE3</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>CPT1B</td>\n",
       "      <td>23291502</td>\n",
       "      <td>0.08</td>\n",
       "      <td>23291502</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>CHKB</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>MAPK8IP2</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>ARSA</td>\n",
       "      <td>19409585</td>\n",
       "      <td>0.08</td>\n",
       "      <td>19409585</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>SHANK3</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>ACR</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>RABL2B</td>\n",
       "      <td>27173419</td>\n",
       "      <td>0.08</td>\n",
       "      <td>27173419</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>429 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Gene  total_coverage  average_coverage  20_total_cvg  \\\n",
       "0           POTEH         3881917              0.08       3881917   \n",
       "1          OR11H1         3881917              0.08       3881917   \n",
       "2          CCT8L2         3881917              0.08       3881917   \n",
       "3            XKR3         7763834              0.08       7763834   \n",
       "4            GAB4         3881917              0.08       3881917   \n",
       "5          IL17RA         7763834              0.08       7763834   \n",
       "6           CECR6         7763834              0.08       7763834   \n",
       "7           CECR5         7763834              0.08       7763834   \n",
       "8           CECR1        23291502              0.08      23291502   \n",
       "9           CECR2         7763834              0.08       7763834   \n",
       "10       SLC25A18         7763834              0.08       7763834   \n",
       "11   LOC101929372         3881917              0.08       3881917   \n",
       "12       ATP6V1E1        11645751              0.08      11645751   \n",
       "13        BCL2L13        42701087              0.08      42701087   \n",
       "14            BID        27173419              0.08      27173419   \n",
       "15         MICAL3        11645751              0.08      11645751   \n",
       "16          PEX26        11645751              0.08      11645751   \n",
       "17          TUBA8         7763834              0.08       7763834   \n",
       "18          USP18         3881917              0.08       3881917   \n",
       "19          DGCR6         3881917              0.08       3881917   \n",
       "20          PRODH         7763834              0.08       7763834   \n",
       "21          DGCR2        15527668              0.08      15527668   \n",
       "22         DGCR14         3881917              0.08       3881917   \n",
       "23          TSSK2         3881917              0.08       3881917   \n",
       "24           GSC2         3881917              0.08       3881917   \n",
       "25        SLC25A1        11645751              0.08      11645751   \n",
       "26         CLTCL1         7763834              0.08       7763834   \n",
       "27           HIRA         3881917              0.08       3881917   \n",
       "28         MRPL40        11645751              0.08      11645751   \n",
       "29       C22orf39         7763834              0.08       7763834   \n",
       "..            ...             ...               ...           ...   \n",
       "399       IL17REL         3881917              0.08       3881917   \n",
       "400          MLC1         7763834              0.08       7763834   \n",
       "401       MOV10L1        15527668              0.08      15527668   \n",
       "402         PANX2         7763834              0.08       7763834   \n",
       "403         TRABD        19409585              0.08      19409585   \n",
       "404       SELENOO         3881917              0.08       3881917   \n",
       "405       TUBGCP6         3881917              0.08       3881917   \n",
       "406        HDAC10         7763834              0.08       7763834   \n",
       "407        MAPK12         7763834              0.08       7763834   \n",
       "408        MAPK11         3881917              0.08       3881917   \n",
       "409        PLXNB2         3881917              0.08       3881917   \n",
       "410       DENND6B         3881917              0.08       3881917   \n",
       "411        PPP6R2        15527668              0.08      15527668   \n",
       "412          SBF1         3881917              0.08       3881917   \n",
       "413          ADM2         3881917              0.08       3881917   \n",
       "414          MIOX         3881917              0.08       3881917   \n",
       "415          LMF2         3881917              0.08       3881917   \n",
       "416        NCAPH2        11645751              0.08      11645751   \n",
       "417          SCO2        15527668              0.08      15527668   \n",
       "418          TYMP        19409585              0.08      19409585   \n",
       "419         ODF3B         3881917              0.08       3881917   \n",
       "420       KLHDC7B         3881917              0.08       3881917   \n",
       "421         SYCE3         3881917              0.08       3881917   \n",
       "422         CPT1B        23291502              0.08      23291502   \n",
       "423          CHKB         3881917              0.08       3881917   \n",
       "424      MAPK8IP2         3881917              0.08       3881917   \n",
       "425          ARSA        19409585              0.08      19409585   \n",
       "426        SHANK3         3881917              0.08       3881917   \n",
       "427           ACR         3881917              0.08       3881917   \n",
       "428        RABL2B        27173419              0.08      27173419   \n",
       "\n",
       "     20_mean_cvg  20_granular_Q1  20_granular_median  20_granular_Q3  \\\n",
       "0           0.08               1                   1               1   \n",
       "1           0.08               1                   1               1   \n",
       "2           0.08               1                   1               1   \n",
       "3           0.08               1                   1               1   \n",
       "4           0.08               1                   1               1   \n",
       "5           0.08               1                   1               1   \n",
       "6           0.08               1                   1               1   \n",
       "7           0.08               1                   1               1   \n",
       "8           0.08               1                   1               1   \n",
       "9           0.08               1                   1               1   \n",
       "10          0.08               1                   1               1   \n",
       "11          0.08               1                   1               1   \n",
       "12          0.08               1                   1               1   \n",
       "13          0.08               1                   1               1   \n",
       "14          0.08               1                   1               1   \n",
       "15          0.08               1                   1               1   \n",
       "16          0.08               1                   1               1   \n",
       "17          0.08               1                   1               1   \n",
       "18          0.08               1                   1               1   \n",
       "19          0.08               1                   1               1   \n",
       "20          0.08               1                   1               1   \n",
       "21          0.08               1                   1               1   \n",
       "22          0.08               1                   1               1   \n",
       "23          0.08               1                   1               1   \n",
       "24          0.08               1                   1               1   \n",
       "25          0.08               1                   1               1   \n",
       "26          0.08               1                   1               1   \n",
       "27          0.08               1                   1               1   \n",
       "28          0.08               1                   1               1   \n",
       "29          0.08               1                   1               1   \n",
       "..           ...             ...                 ...             ...   \n",
       "399         0.08               1                   1               1   \n",
       "400         0.08               1                   1               1   \n",
       "401         0.08               1                   1               1   \n",
       "402         0.08               1                   1               1   \n",
       "403         0.08               1                   1               1   \n",
       "404         0.08               1                   1               1   \n",
       "405         0.08               1                   1               1   \n",
       "406         0.08               1                   1               1   \n",
       "407         0.08               1                   1               1   \n",
       "408         0.08               1                   1               1   \n",
       "409         0.08               1                   1               1   \n",
       "410         0.08               1                   1               1   \n",
       "411         0.08               1                   1               1   \n",
       "412         0.08               1                   1               1   \n",
       "413         0.08               1                   1               1   \n",
       "414         0.08               1                   1               1   \n",
       "415         0.08               1                   1               1   \n",
       "416         0.08               1                   1               1   \n",
       "417         0.08               1                   1               1   \n",
       "418         0.08               1                   1               1   \n",
       "419         0.08               1                   1               1   \n",
       "420         0.08               1                   1               1   \n",
       "421         0.08               1                   1               1   \n",
       "422         0.08               1                   1               1   \n",
       "423         0.08               1                   1               1   \n",
       "424         0.08               1                   1               1   \n",
       "425         0.08               1                   1               1   \n",
       "426         0.08               1                   1               1   \n",
       "427         0.08               1                   1               1   \n",
       "428         0.08               1                   1               1   \n",
       "\n",
       "     20_%_above_15  \n",
       "0              0.0  \n",
       "1              0.0  \n",
       "2              0.0  \n",
       "3              0.0  \n",
       "4              0.0  \n",
       "5              0.0  \n",
       "6              0.0  \n",
       "7              0.0  \n",
       "8              0.0  \n",
       "9              0.0  \n",
       "10             0.0  \n",
       "11             0.0  \n",
       "12             0.0  \n",
       "13             0.0  \n",
       "14             0.0  \n",
       "15             0.0  \n",
       "16             0.0  \n",
       "17             0.0  \n",
       "18             0.0  \n",
       "19             0.0  \n",
       "20             0.0  \n",
       "21             0.0  \n",
       "22             0.0  \n",
       "23             0.0  \n",
       "24             0.0  \n",
       "25             0.0  \n",
       "26             0.0  \n",
       "27             0.0  \n",
       "28             0.0  \n",
       "29             0.0  \n",
       "..             ...  \n",
       "399            0.0  \n",
       "400            0.0  \n",
       "401            0.0  \n",
       "402            0.0  \n",
       "403            0.0  \n",
       "404            0.0  \n",
       "405            0.0  \n",
       "406            0.0  \n",
       "407            0.0  \n",
       "408            0.0  \n",
       "409            0.0  \n",
       "410            0.0  \n",
       "411            0.0  \n",
       "412            0.0  \n",
       "413            0.0  \n",
       "414            0.0  \n",
       "415            0.0  \n",
       "416            0.0  \n",
       "417            0.0  \n",
       "418            0.0  \n",
       "419            0.0  \n",
       "420            0.0  \n",
       "421            0.0  \n",
       "422            0.0  \n",
       "423            0.0  \n",
       "424            0.0  \n",
       "425            0.0  \n",
       "426            0.0  \n",
       "427            0.0  \n",
       "428            0.0  \n",
       "\n",
       "[429 rows x 9 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DEPTH_DIR='/hackathon/Hackathon_Project_4/ENCODE_DATA_GM12878/COMPLETED/DepthOfCoverage_test/'\n",
    "jp2=lambda x: os.path.join(DEPTH_DIR,x)\n",
    "\n",
    "pd.read_table(jp2('ENCFF000ATH_chr22_ChIP-seq_rg.depth.sample_gene_summary'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>depth&gt;=0</th>\n",
       "      <th>depth&gt;=1</th>\n",
       "      <th>depth&gt;=2</th>\n",
       "      <th>depth&gt;=3</th>\n",
       "      <th>depth&gt;=4</th>\n",
       "      <th>depth&gt;=5</th>\n",
       "      <th>depth&gt;=6</th>\n",
       "      <th>depth&gt;=7</th>\n",
       "      <th>depth&gt;=8</th>\n",
       "      <th>...</th>\n",
       "      <th>depth&gt;=3266</th>\n",
       "      <th>depth&gt;=3408</th>\n",
       "      <th>depth&gt;=3556</th>\n",
       "      <th>depth&gt;=3711</th>\n",
       "      <th>depth&gt;=3872</th>\n",
       "      <th>depth&gt;=4041</th>\n",
       "      <th>depth&gt;=4217</th>\n",
       "      <th>depth&gt;=4400</th>\n",
       "      <th>depth&gt;=4591</th>\n",
       "      <th>depth&gt;=5000</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>At_least_1_samples</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows  203 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Unnamed: 0  depth>=0  depth>=1  depth>=2  depth>=3  depth>=4  \\\n",
       "0  At_least_1_samples         1         0         0         0         0   \n",
       "\n",
       "   depth>=5  depth>=6  depth>=7  depth>=8     ...       depth>=3266  \\\n",
       "0         0         0         0         0     ...                 0   \n",
       "\n",
       "   depth>=3408  depth>=3556  depth>=3711  depth>=3872  depth>=4041  \\\n",
       "0            0            0            0            0            0   \n",
       "\n",
       "   depth>=4217  depth>=4400  depth>=4591  depth>=5000  \n",
       "0            0            0            0            0  \n",
       "\n",
       "[1 rows x 203 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table(jp2('ENCFF000ATH_chr22_ChIP-seq_rg.depth.sample_interval_statistics'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Target</th>\n",
       "      <th>total_coverage</th>\n",
       "      <th>average_coverage</th>\n",
       "      <th>20_total_cvg</th>\n",
       "      <th>20_mean_cvg</th>\n",
       "      <th>20_granular_Q1</th>\n",
       "      <th>20_granular_median</th>\n",
       "      <th>20_granular_Q3</th>\n",
       "      <th>20_%_above_15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr22:1-51304566</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Target  total_coverage  average_coverage  20_total_cvg  \\\n",
       "0  chr22:1-51304566         3881917              0.08       3881917   \n",
       "\n",
       "   20_mean_cvg  20_granular_Q1  20_granular_median  20_granular_Q3  \\\n",
       "0         0.08               1                   1               1   \n",
       "\n",
       "   20_%_above_15  \n",
       "0            0.0  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table(jp2('ENCFF000ATH_chr22_ChIP-seq_rg.depth.sample_interval_summary'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Source_of_reads</th>\n",
       "      <th>from_0_to_1)</th>\n",
       "      <th>from_1_to_2)</th>\n",
       "      <th>from_2_to_3)</th>\n",
       "      <th>from_3_to_4)</th>\n",
       "      <th>from_4_to_5)</th>\n",
       "      <th>from_5_to_6)</th>\n",
       "      <th>from_6_to_7)</th>\n",
       "      <th>from_7_to_8)</th>\n",
       "      <th>from_8_to_9)</th>\n",
       "      <th>...</th>\n",
       "      <th>from_3266_to_3408)</th>\n",
       "      <th>from_3408_to_3556)</th>\n",
       "      <th>from_3556_to_3711)</th>\n",
       "      <th>from_3711_to_3872)</th>\n",
       "      <th>from_3872_to_4041)</th>\n",
       "      <th>from_4041_to_4217)</th>\n",
       "      <th>from_4217_to_4400)</th>\n",
       "      <th>from_4400_to_4591)</th>\n",
       "      <th>from_4591_to_5000)</th>\n",
       "      <th>from_5000_to_inf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sample_20</td>\n",
       "      <td>48665852</td>\n",
       "      <td>1887281</td>\n",
       "      <td>454094</td>\n",
       "      <td>176676</td>\n",
       "      <td>74039</td>\n",
       "      <td>29223</td>\n",
       "      <td>11011</td>\n",
       "      <td>4045</td>\n",
       "      <td>1591</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows  203 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Source_of_reads  from_0_to_1)  from_1_to_2)  from_2_to_3)  from_3_to_4)  \\\n",
       "0       sample_20      48665852       1887281        454094        176676   \n",
       "\n",
       "   from_4_to_5)  from_5_to_6)  from_6_to_7)  from_7_to_8)  from_8_to_9)  \\\n",
       "0         74039         29223         11011          4045          1591   \n",
       "\n",
       "         ...         from_3266_to_3408)  from_3408_to_3556)  \\\n",
       "0        ...                          0                   0   \n",
       "\n",
       "   from_3556_to_3711)  from_3711_to_3872)  from_3872_to_4041)  \\\n",
       "0                   0                   0                   0   \n",
       "\n",
       "   from_4041_to_4217)  from_4217_to_4400)  from_4400_to_4591)  \\\n",
       "0                   0                   0                   0   \n",
       "\n",
       "   from_4591_to_5000)  from_5000_to_inf  \n",
       "0                   0                 0  \n",
       "\n",
       "[1 rows x 203 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table(jp2('ENCFF000ATH_chr22_ChIP-seq_rg.depth.sample_statistics'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>total</th>\n",
       "      <th>mean</th>\n",
       "      <th>granular_third_quartile</th>\n",
       "      <th>granular_median</th>\n",
       "      <th>granular_first_quartile</th>\n",
       "      <th>%_bases_above_15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Total</td>\n",
       "      <td>3881917</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id    total  mean  granular_third_quartile  granular_median  \\\n",
       "0        20  3881917  0.08                      1.0              1.0   \n",
       "1     Total  3881917  0.08                      NaN              NaN   \n",
       "\n",
       "   granular_first_quartile  %_bases_above_15  \n",
       "0                      1.0               0.0  \n",
       "1                      NaN               NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_table(jp2('ENCFF000ATH_chr22_ChIP-seq_rg.depth.sample_summary'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See ``GATK DepthOfCoverage`` [manual](https://software.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_coverage_DepthOfCoverage.php#--calculateCoverageOverGenes).\n",
    "\n",
    "``GATK DepthOfCoverage`` requires:\n",
    "- an index reference fasta\n",
    "- indexed and read grouped bam files\n",
    "- [optinal] gene files from [UCSC genome ftp sites](http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#-l INFO --omitDepthOutputAtEachBase --omitLocusTable \\\n",
    "gatk -T DepthOfCoverage \\\n",
    "-R \"/hackathon/Hackathon_Project_4/REFERENCE_GENOME/hg19_ENCODE.fa\" \\\n",
    "-I \"/hackathon/Hackathon_Project_4/ENCODE_DATA_GM12878/COMPLETED/rg_bams/ENCFF000ATH_chr22_ChIP-seq_rg.bam\" \\\n",
    "-L chr22 \\\n",
    "--minBaseQuality 20 \\\n",
    "--minMappingQuality 10 \\\n",
    "--start 1 \\\n",
    "--stop 5000 \\\n",
    "--nBins 200 \\\n",
    "--includeRefNSites \\\n",
    "--countType COUNT_FRAGMENTS \\\n",
    "--calculateCoverageOverGenes \"/hackathon/Hackathon_Project_4/REFERENCE_GENOME/refGene_chr22_sorted.txt\" \\\n",
    "-o \"/hackathon/Hackathon_Project_4/ENCODE_DATA_GM12878/COMPLETED/DepthOfCoverage_test/ENCFF000ATH_chr22_ChIP-seq_rg.depth\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>Total_Depth</th>\n",
       "      <th>Average_Depth_sample</th>\n",
       "      <th>Depth_for_20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr22:1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Locus  Total_Depth  Average_Depth_sample  Depth_for_20\n",
       "0  chr22:1            0                   0.0             0"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ENCFF000ATH_chr22_depth=pd.read_table(jp2('ENCFF000ATH_chr22_ChIP-seq_rg.depth'))\n",
    "ENCFF000ATH_chr22_depth[:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>Total_Depth</th>\n",
       "      <th>Average_Depth_sample</th>\n",
       "      <th>Depth_for_20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr22:3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr22:4</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr22:5</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr22:6</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr22:7</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr22:8</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr22:9</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr22:10</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr22:11</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr22:12</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr22:13</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr22:14</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr22:15</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr22:16</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr22:17</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr22:18</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr22:19</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr22:20</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr22:21</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr22:22</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr22:23</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr22:24</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr22:25</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr22:26</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr22:27</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr22:28</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr22:29</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr22:30</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr22:31</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr22:32</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304536</th>\n",
       "      <td>chr22:51304537</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304537</th>\n",
       "      <td>chr22:51304538</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304538</th>\n",
       "      <td>chr22:51304539</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304539</th>\n",
       "      <td>chr22:51304540</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304540</th>\n",
       "      <td>chr22:51304541</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304541</th>\n",
       "      <td>chr22:51304542</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304542</th>\n",
       "      <td>chr22:51304543</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304543</th>\n",
       "      <td>chr22:51304544</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304544</th>\n",
       "      <td>chr22:51304545</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304545</th>\n",
       "      <td>chr22:51304546</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304546</th>\n",
       "      <td>chr22:51304547</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304547</th>\n",
       "      <td>chr22:51304548</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304548</th>\n",
       "      <td>chr22:51304549</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304549</th>\n",
       "      <td>chr22:51304550</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304550</th>\n",
       "      <td>chr22:51304551</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304551</th>\n",
       "      <td>chr22:51304552</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304552</th>\n",
       "      <td>chr22:51304553</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304553</th>\n",
       "      <td>chr22:51304554</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304554</th>\n",
       "      <td>chr22:51304555</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304555</th>\n",
       "      <td>chr22:51304556</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304556</th>\n",
       "      <td>chr22:51304557</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304557</th>\n",
       "      <td>chr22:51304558</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304558</th>\n",
       "      <td>chr22:51304559</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304559</th>\n",
       "      <td>chr22:51304560</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304560</th>\n",
       "      <td>chr22:51304561</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304561</th>\n",
       "      <td>chr22:51304562</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304562</th>\n",
       "      <td>chr22:51304563</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304563</th>\n",
       "      <td>chr22:51304564</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304564</th>\n",
       "      <td>chr22:51304565</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51304565</th>\n",
       "      <td>chr22:51304566</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>51304564 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Locus  Total_Depth  Average_Depth_sample  Depth_for_20\n",
       "2                chr22:3            0                   0.0             0\n",
       "3                chr22:4            0                   0.0             0\n",
       "4                chr22:5            0                   0.0             0\n",
       "5                chr22:6            0                   0.0             0\n",
       "6                chr22:7            0                   0.0             0\n",
       "7                chr22:8            0                   0.0             0\n",
       "8                chr22:9            0                   0.0             0\n",
       "9               chr22:10            0                   0.0             0\n",
       "10              chr22:11            0                   0.0             0\n",
       "11              chr22:12            0                   0.0             0\n",
       "12              chr22:13            0                   0.0             0\n",
       "13              chr22:14            0                   0.0             0\n",
       "14              chr22:15            0                   0.0             0\n",
       "15              chr22:16            0                   0.0             0\n",
       "16              chr22:17            0                   0.0             0\n",
       "17              chr22:18            0                   0.0             0\n",
       "18              chr22:19            0                   0.0             0\n",
       "19              chr22:20            0                   0.0             0\n",
       "20              chr22:21            0                   0.0             0\n",
       "21              chr22:22            0                   0.0             0\n",
       "22              chr22:23            0                   0.0             0\n",
       "23              chr22:24            0                   0.0             0\n",
       "24              chr22:25            0                   0.0             0\n",
       "25              chr22:26            0                   0.0             0\n",
       "26              chr22:27            0                   0.0             0\n",
       "27              chr22:28            0                   0.0             0\n",
       "28              chr22:29            0                   0.0             0\n",
       "29              chr22:30            0                   0.0             0\n",
       "30              chr22:31            0                   0.0             0\n",
       "31              chr22:32            0                   0.0             0\n",
       "...                  ...          ...                   ...           ...\n",
       "51304536  chr22:51304537            0                   0.0             0\n",
       "51304537  chr22:51304538            0                   0.0             0\n",
       "51304538  chr22:51304539            0                   0.0             0\n",
       "51304539  chr22:51304540            0                   0.0             0\n",
       "51304540  chr22:51304541            0                   0.0             0\n",
       "51304541  chr22:51304542            0                   0.0             0\n",
       "51304542  chr22:51304543            0                   0.0             0\n",
       "51304543  chr22:51304544            0                   0.0             0\n",
       "51304544  chr22:51304545            0                   0.0             0\n",
       "51304545  chr22:51304546            0                   0.0             0\n",
       "51304546  chr22:51304547            0                   0.0             0\n",
       "51304547  chr22:51304548            0                   0.0             0\n",
       "51304548  chr22:51304549            0                   0.0             0\n",
       "51304549  chr22:51304550            0                   0.0             0\n",
       "51304550  chr22:51304551            0                   0.0             0\n",
       "51304551  chr22:51304552            0                   0.0             0\n",
       "51304552  chr22:51304553            0                   0.0             0\n",
       "51304553  chr22:51304554            0                   0.0             0\n",
       "51304554  chr22:51304555            0                   0.0             0\n",
       "51304555  chr22:51304556            0                   0.0             0\n",
       "51304556  chr22:51304557            0                   0.0             0\n",
       "51304557  chr22:51304558            0                   0.0             0\n",
       "51304558  chr22:51304559            0                   0.0             0\n",
       "51304559  chr22:51304560            0                   0.0             0\n",
       "51304560  chr22:51304561            0                   0.0             0\n",
       "51304561  chr22:51304562            0                   0.0             0\n",
       "51304562  chr22:51304563            0                   0.0             0\n",
       "51304563  chr22:51304564            0                   0.0             0\n",
       "51304564  chr22:51304565            0                   0.0             0\n",
       "51304565  chr22:51304566            0                   0.0             0\n",
       "\n",
       "[51304564 rows x 4 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ENCFF000ATH_chr22_depth[2:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ENCFF000ATH_chr22_depth_moreThanTen = ENCFF000ATH_chr22_depth[ENCFF000ATH_chr22_depth['Total_Depth']>=10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>Total_Depth</th>\n",
       "      <th>Average_Depth_sample</th>\n",
       "      <th>Depth_for_20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17698346</th>\n",
       "      <td>chr22:17698347</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698347</th>\n",
       "      <td>chr22:17698348</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698349</th>\n",
       "      <td>chr22:17698350</td>\n",
       "      <td>10</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Locus  Total_Depth  Average_Depth_sample  Depth_for_20\n",
       "17698346  chr22:17698347           10                  10.0            10\n",
       "17698347  chr22:17698348           10                  10.0            10\n",
       "17698349  chr22:17698350           10                  10.0            10"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ENCFF000ATH_chr22_depth_moreThanTen[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Locus</th>\n",
       "      <th>Depth_for_20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17698346</th>\n",
       "      <td>chr22:17698347</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698347</th>\n",
       "      <td>chr22:17698348</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698349</th>\n",
       "      <td>chr22:17698350</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698350</th>\n",
       "      <td>chr22:17698351</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698351</th>\n",
       "      <td>chr22:17698352</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17698352</th>\n",
       "      <td>chr22:17698353</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18120985</th>\n",
       "      <td>chr22:18120986</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18120986</th>\n",
       "      <td>chr22:18120987</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18120987</th>\n",
       "      <td>chr22:18120988</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18120988</th>\n",
       "      <td>chr22:18120989</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18120989</th>\n",
       "      <td>chr22:18120990</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18254754</th>\n",
       "      <td>chr22:18254755</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18254757</th>\n",
       "      <td>chr22:18254758</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18254759</th>\n",
       "      <td>chr22:18254760</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18261291</th>\n",
       "      <td>chr22:18261292</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18261294</th>\n",
       "      <td>chr22:18261295</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18261295</th>\n",
       "      <td>chr22:18261296</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279793</th>\n",
       "      <td>chr22:19279794</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279794</th>\n",
       "      <td>chr22:19279795</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279795</th>\n",
       "      <td>chr22:19279796</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279797</th>\n",
       "      <td>chr22:19279798</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279799</th>\n",
       "      <td>chr22:19279800</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279801</th>\n",
       "      <td>chr22:19279802</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279802</th>\n",
       "      <td>chr22:19279803</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279806</th>\n",
       "      <td>chr22:19279807</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19279890</th>\n",
       "      <td>chr22:19279891</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19930559</th>\n",
       "      <td>chr22:19930560</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19930688</th>\n",
       "      <td>chr22:19930689</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19930690</th>\n",
       "      <td>chr22:19930691</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19930692</th>\n",
       "      <td>chr22:19930693</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42466245</th>\n",
       "      <td>chr22:42466246</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42466247</th>\n",
       "      <td>chr22:42466248</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009443</th>\n",
       "      <td>chr22:43009444</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009444</th>\n",
       "      <td>chr22:43009445</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009446</th>\n",
       "      <td>chr22:43009447</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009447</th>\n",
       "      <td>chr22:43009448</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009448</th>\n",
       "      <td>chr22:43009449</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009449</th>\n",
       "      <td>chr22:43009450</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43009451</th>\n",
       "      <td>chr22:43009452</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631207</th>\n",
       "      <td>chr22:45631208</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631208</th>\n",
       "      <td>chr22:45631209</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631209</th>\n",
       "      <td>chr22:45631210</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631210</th>\n",
       "      <td>chr22:45631211</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631211</th>\n",
       "      <td>chr22:45631212</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631212</th>\n",
       "      <td>chr22:45631213</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631213</th>\n",
       "      <td>chr22:45631214</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631214</th>\n",
       "      <td>chr22:45631215</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631215</th>\n",
       "      <td>chr22:45631216</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631216</th>\n",
       "      <td>chr22:45631217</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45631217</th>\n",
       "      <td>chr22:45631218</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46646958</th>\n",
       "      <td>chr22:46646959</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48494597</th>\n",
       "      <td>chr22:48494598</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48494598</th>\n",
       "      <td>chr22:48494599</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059148</th>\n",
       "      <td>chr22:51059149</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059149</th>\n",
       "      <td>chr22:51059150</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059150</th>\n",
       "      <td>chr22:51059151</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059152</th>\n",
       "      <td>chr22:51059153</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059154</th>\n",
       "      <td>chr22:51059155</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059155</th>\n",
       "      <td>chr22:51059156</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51059160</th>\n",
       "      <td>chr22:51059161</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Locus  Depth_for_20\n",
       "17698346  chr22:17698347            10\n",
       "17698347  chr22:17698348            10\n",
       "17698349  chr22:17698350            10\n",
       "17698350  chr22:17698351            10\n",
       "17698351  chr22:17698352            11\n",
       "17698352  chr22:17698353            10\n",
       "18120985  chr22:18120986            10\n",
       "18120986  chr22:18120987            10\n",
       "18120987  chr22:18120988            10\n",
       "18120988  chr22:18120989            10\n",
       "18120989  chr22:18120990            10\n",
       "18254754  chr22:18254755            10\n",
       "18254757  chr22:18254758            11\n",
       "18254759  chr22:18254760            10\n",
       "18261291  chr22:18261292            10\n",
       "18261294  chr22:18261295            10\n",
       "18261295  chr22:18261296            10\n",
       "19279793  chr22:19279794            10\n",
       "19279794  chr22:19279795            10\n",
       "19279795  chr22:19279796            10\n",
       "19279797  chr22:19279798            10\n",
       "19279799  chr22:19279800            10\n",
       "19279801  chr22:19279802            10\n",
       "19279802  chr22:19279803            10\n",
       "19279806  chr22:19279807            10\n",
       "19279890  chr22:19279891            10\n",
       "19930559  chr22:19930560            10\n",
       "19930688  chr22:19930689            10\n",
       "19930690  chr22:19930691            10\n",
       "19930692  chr22:19930693            10\n",
       "...                  ...           ...\n",
       "42466245  chr22:42466246            10\n",
       "42466247  chr22:42466248            10\n",
       "43009443  chr22:43009444            11\n",
       "43009444  chr22:43009445            10\n",
       "43009446  chr22:43009447            11\n",
       "43009447  chr22:43009448            11\n",
       "43009448  chr22:43009449            11\n",
       "43009449  chr22:43009450            10\n",
       "43009451  chr22:43009452            10\n",
       "45631207  chr22:45631208            10\n",
       "45631208  chr22:45631209            10\n",
       "45631209  chr22:45631210            11\n",
       "45631210  chr22:45631211            11\n",
       "45631211  chr22:45631212            10\n",
       "45631212  chr22:45631213            11\n",
       "45631213  chr22:45631214            11\n",
       "45631214  chr22:45631215            11\n",
       "45631215  chr22:45631216            11\n",
       "45631216  chr22:45631217            11\n",
       "45631217  chr22:45631218            10\n",
       "46646958  chr22:46646959            10\n",
       "48494597  chr22:48494598            10\n",
       "48494598  chr22:48494599            10\n",
       "51059148  chr22:51059149            10\n",
       "51059149  chr22:51059150            11\n",
       "51059150  chr22:51059151            11\n",
       "51059152  chr22:51059153            10\n",
       "51059154  chr22:51059155            10\n",
       "51059155  chr22:51059156            10\n",
       "51059160  chr22:51059161            10\n",
       "\n",
       "[193 rows x 2 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ENCFF000ATH_chr22_depth_moreThanTen[['Locus','Depth_for_20']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "CHROMOSOME=\"chr22\" # change this to any chromosome you like\n",
    "BASE_PATH='/hackathon/Hackathon_Project_4/'\n",
    "\n",
    "#INDEXED_RG_BAMPATHS=\"/hackathon/Hackathon_Project_4/ENCODE_DATA_GM12878/COMPLETED/*_rg*bai\" # TODO: change this to bam when everything is indexed\n",
    "#REFERENCE=\"/hackathon/Hackathon_Project_4/REFERENCE_GENOME/hg19_ENCODE.fa\"\n",
    "#GENEFILE=\"/hackathon/Hackathon_Project_4/REFERENCE_GENOME/refGene_\"$CHROMOSOME\"_sorted.txt\"\n",
    "#OUTPUT_DIRECTORY=\"/hackathon/Hackathon_Project_4/ENCODE_DATA_GM12878/COMPLETED/DepthOfCoverage\"\n",
    "\n",
    "REFERENCE=$BASE_PATH\"REFERENCE_GENOME/hg19_ENCODE.fa\"\n",
    "GENEFILE=$BASE_PATH\"REFERENCE_GENOME/refGene_\"$CHROMOSOME\"_sorted.txt\"\n",
    "\n",
    "BAM_PATHS=$BASE_PATH\"ENCODE_DATA_GM12878/COMPLETED/rg_bams/\"\n",
    "#BAM_PATHS=$BASE_PATH\"ENCODE_DATA_GM12878/COMPLETED/DepthOfCoverage/\"\n",
    "OUTPUT_DIRECTORY=$BASE_PATH\"ENCODE_DATA_GM12878/COMPLETED/DepthOfCoverage\"\n",
    "BAMS=$BAM_PATHS\"*_rg*bai\" # Get indexed rg bam files (Run \"readGroup.sh\")\n",
    "#BAMS=$BAM_PATHS\"ENCFF000WCN*_rg*bai\" # Get indexed rg bam files (Run \"readGroup.sh\")\n",
    "#BAMS=$BAM_PATHS\"ENCFF000WCQ*_rg*bai\" # Get indexed rg bam files (Run \"readGroup.sh\")\n",
    "#BAMS=$BAM_PATHS\"ENCFF000WD*_rg*bai\" # Get indexed rg bam files (Run \"readGroup.sh\")\n",
    "#mkdir $OUTPUT_DIRECTORY\n",
    "\n",
    "DEPTH_COMMANDS_FILE=${OUTPUT_DIRECTORY}/\"depth_commands.sh\"\n",
    "\n",
    "for bam in $(ls $BAMS);\n",
    "do\n",
    "    # change the regex extension replacement \n",
    "    #DEPTH_COMMAND=\"gatk -T DepthOfCoverage -R $REFERENCE -I $(echo $bam|sed 's/.bai//g') -L $CHROMOSOME -l INFO --omitDepthOutputAtEachBase --omitLocusTable --minBaseQuality 20 --minMappingQuality 20 --start 1 --stop 5000 --nBins 200 --includeRefNSites --countType COUNT_FRAGMENTS --calculateCoverageOverGenes $GENEFILE -o ${OUTPUT_DIRECTORY}/$(basename $bam|sed 's/.bam.bai/.depth/g')\"\n",
    "    DEPTH_COMMAND=\"gatk -T DepthOfCoverage -R $REFERENCE -I $(echo $bam|sed 's/.bai//g') -L $CHROMOSOME --minBaseQuality 20 --minMappingQuality 10 --start 1 --stop 5000 --nBins 200 --includeRefNSites --countType COUNT_FRAGMENTS --calculateCoverageOverGenes $GENEFILE -o ${OUTPUT_DIRECTORY}/$(basename $bam|sed 's/.bam.bai/.depth/g')\"\n",
    "    $DEPTH_COMMAND\n",
    "    #echo $DEPTH_COMMAND\n",
    "done >> $DEPTH_COMMANDS_FILE\n",
    "\n",
    "#head ${DEPTH_COMMANDS_FILE}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "GATK DepthOfCove WITH A SINGLE MERGED BAM\n",
    "=========================================\n",
    "\n",
    "This merged bam ``/hackathon/Hackathon_Project_4/VariantCall_HAPLOTYPE/merged_rg.chr22.bam`` is constructed with ENCFF000ARG and ENCFF000ARI. The read groups are tagged so that we know the source."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#REFERENCE=\"/hackathon/Hackathon_Project_4/REFERENCE_GENOME/hg19.fa\"\n",
    "bam=\"/hackathon/Hackathon_Project_4/VariantCall_HAPLOTYPE/merged_rg.chr22.bam\" # TODO: change this to bam when everything is indexed\n",
    "CHROMOSOME=\"chr22\" # change this to any chromosome you like\n",
    "BASE_PATH='/hackathon/Hackathon_Project_4/'\n",
    "BAM_PATHS=$BASE_PATH\"VariantCall_HAPLOTYPE/\"\n",
    "OUTPUT_DIRECTORY=$BAM_PATHS\"DepthOfCoverage\"\n",
    "mkdir $OUTPUT_DIRECTORY\n",
    "\n",
    "REFERENCE=$BASE_PATH\"REFERENCE_GENOME/hg19_ENCODE.fa\"\n",
    "GENEFILE=$BASE_PATH\"REFERENCE_GENOME/refGene_\"$CHROMOSOME\"_sorted.txt\"\n",
    "\n",
    "gatk -T DepthOfCoverage -R $REFERENCE -I $(echo $bam|sed 's/.bai//g') -L $CHROMOSOME -l INFO --omitDepthOutputAtEachBase --omitLocusTable --minBaseQuality 20 --minMappingQuality 20 --start 1 --stop 5000 --nBins 200 --includeRefNSites --countType COUNT_FRAGMENTS --calculateCoverageOverGenes $GENEFILE -o ${OUTPUT_DIRECTORY}/$(basename $bam|sed 's/.bam.bai/.depth/g')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
